### PR TITLE
comparison result unused

### PR DESCRIPTION
**Pull request by jzhangg85**

### CODE REVIEW

File: helpers/clang.py [hunk 1]
```diff
@@ -4,6 +4,13 @@
 def help(lines):
 
     # $ clang foo.c
+    # foo.c:12:19: error: relational comparison result unused [-Weeror,-Wunused-comparison]
+    matches = match(r"relational comparison result unused", lines[0])
+    if matches: 
+        response = ["Looks like you are trying to make a comparison that has no effect probably due to a syntax error inside of your for loop."]
```
Comment (dlloyd09) [inline helpers/clang.py]: Is a for loop always the place where this error would be given by clang?

File: helpers/clang.py [hunk 2]
```diff
@@ -473,6 +466,13 @@ def help(lines):
         return (1, response)
 
     # $ clang foo.c
+    # foo.c:12:19: error: relational comparison result unused [-Weeror,-Wunused-comparison]
+    matches = match(r"relational comparison result unused", lines[0])
+    if matches: 
+        response = ["Looks like you are trying to make a comparison that has no effect probably due to a syntax error around the conditional comparison or two conflicting comparisons called due to the lack of an operator such as && or ||."]
```
Comment (dmalan) [inline helpers/clang.py]: Hm, not sure this advice will be all that helpful to a student; it's pretty complex-sounding and also very guessy. Consider the examples at https://github.com/cs50/help50/issues/126.

Perhaps better would be, e.g.:

```
Looks like you're comparing two values on line {} of `{}` but not using the result?
```

And if `len(lines) >= 3`, we should include `lines[1]` and `lines[2]` in the return value, as with:

```
return (lines[0:3], response);
```

New syntax as of this weekend.

CC @brianyu28

File: helpers/clang.py [hunk 3]
```diff
@@ -473,6 +466,13 @@ def help(lines):
         return (1, response)
 
     # $ clang foo.c
+    # foo.c:12:19: error: relational comparison result unused [-Weeror,-Wunused-comparison]
```
Comment (dmalan) [inline helpers/clang.py]: Do add more of the example here, as in https://github.com/cs50/help50/issues/126. Else we'll never remember what kind of code this is referring to.

**Comment by dmalan** [comment]
https://github.com/cs50/help50/commit/9b640b41cc0d94bf32888abaa9a4fdf418b62e2b

**Pull request closed without merging**